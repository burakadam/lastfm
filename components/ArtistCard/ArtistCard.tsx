import Image from 'next/image';
import Link from 'next/link';

export interface IArtistCardProps {
  name: string;
  image: string;
  playcount: string;
  listeners: string;
}

const ArtistCard = ({
  name,
  image,
  playcount,
  listeners,
}: IArtistCardProps) => (
  <Link
    href={`/detail/${name.split(' ').join('').toLowerCase()}`}
    className='border p-2 flex mb-2'
  >
    <div className='mr-2'>
      <Image
        src={image}
        alt={name}
        width={500}
        height={500}
        className='w-28 h-28'
      />
    </div>
    <div className='flex-col flex justify-between'>
      <h3 className='font-bold'>
        Artist : <span className='italic'>{name}</span>
      </h3>
      <div className='text-sm text-gray-500'>
        <p>Listeners : {listeners}</p>
        <p>Play Count : {playcount}</p>
      </div>
    </div>
  </Link>
);

export default ArtistCard;
