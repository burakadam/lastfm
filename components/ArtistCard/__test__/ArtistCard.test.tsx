import { render } from '@testing-library/react';

import ArtistCard from '../ArtistCard';

const MOCK_ARTIST = {
  name: 'The Weeknd',
  playcount: '378836644',
  listeners: '2940862',
  mbid: 'c8b03190-306c-4120-bb0b-6f2ebfc06ea9',
  url: 'https://www.last.fm/music/The+Weeknd',
  streamable: '0',
  image:
    'https://lastfm.freetls.fastly.net/i/u/64s/2a96cbd8b46e442fc41c2b86b821562f.png',
};

describe('<ArtistCard> spec', () => {
  let CARD: any;

  beforeEach(() => {
    CARD = render(
      <ArtistCard
        name={MOCK_ARTIST.name}
        playcount={MOCK_ARTIST.playcount}
        listeners={MOCK_ARTIST.listeners}
        image={MOCK_ARTIST.image}
      />
    );
  });

  it('render a artist card component', () => expect(CARD).toMatchSnapshot());

  it('render artist name', () =>
    expect(CARD.getByText(MOCK_ARTIST.name)).toBeTruthy());

  it('render artist playcount', () =>
    expect(
      CARD.getByText(`Play Count : ${MOCK_ARTIST.playcount}`)
    ).toBeTruthy());

  it('render artist listeners', () =>
    expect(
      CARD.getByText(`Listeners : ${MOCK_ARTIST.listeners}`)
    ).toBeTruthy());
});
