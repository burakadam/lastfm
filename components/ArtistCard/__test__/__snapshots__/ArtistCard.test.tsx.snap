// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArtistCard> spec render a artist card component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="border p-2 flex mb-2"
        href="/detail/theweeknd"
      >
        <div
          class="mr-2"
        >
          <img
            alt="The Weeknd"
            class="w-28 h-28"
            data-nimg="1"
            decoding="async"
            height="500"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Flastfm.freetls.fastly.net%2Fi%2Fu%2F64s%2F2a96cbd8b46e442fc41c2b86b821562f.png&w=1080&q=75"
            srcset="/_next/image?url=https%3A%2F%2Flastfm.freetls.fastly.net%2Fi%2Fu%2F64s%2F2a96cbd8b46e442fc41c2b86b821562f.png&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Flastfm.freetls.fastly.net%2Fi%2Fu%2F64s%2F2a96cbd8b46e442fc41c2b86b821562f.png&w=1080&q=75 2x"
            style="color: transparent;"
            width="500"
          />
        </div>
        <div
          class="flex-col flex justify-between"
        >
          <h3
            class="font-bold"
          >
            Artist : 
            <span
              class="italic"
            >
              The Weeknd
            </span>
          </h3>
          <div
            class="text-sm text-gray-500"
          >
            <p>
              Listeners : 
              2940862
            </p>
            <p>
              Play Count : 
              378836644
            </p>
          </div>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="border p-2 flex mb-2"
      href="/detail/theweeknd"
    >
      <div
        class="mr-2"
      >
        <img
          alt="The Weeknd"
          class="w-28 h-28"
          data-nimg="1"
          decoding="async"
          height="500"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Flastfm.freetls.fastly.net%2Fi%2Fu%2F64s%2F2a96cbd8b46e442fc41c2b86b821562f.png&w=1080&q=75"
          srcset="/_next/image?url=https%3A%2F%2Flastfm.freetls.fastly.net%2Fi%2Fu%2F64s%2F2a96cbd8b46e442fc41c2b86b821562f.png&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Flastfm.freetls.fastly.net%2Fi%2Fu%2F64s%2F2a96cbd8b46e442fc41c2b86b821562f.png&w=1080&q=75 2x"
          style="color: transparent;"
          width="500"
        />
      </div>
      <div
        class="flex-col flex justify-between"
      >
        <h3
          class="font-bold"
        >
          Artist : 
          <span
            class="italic"
          >
            The Weeknd
          </span>
        </h3>
        <div
          class="text-sm text-gray-500"
        >
          <p>
            Listeners : 
            2940862
          </p>
          <p>
            Play Count : 
            378836644
          </p>
        </div>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
